metricbeat/module/kubernetes/state_container/state_container.go

```diff
@@ -132,9 +130,19 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 				event["cpu.limit.nanocores"] = limitCores * nanocores
 			}
 		}
-	}
 
-	return events, err
+		moduleFields := event[mb.ModuleDataKey].(common.MapStr)
```

Comment:
Could you add an `ok` check to make sure we don't get a panic in case either moduleDataKey does not exist or the type conversion fails?

Reply:
Second note: I think it's the right thing to do for now. Long term we should find a nicer solution for it.

Reply:
If the conversion fails, `moduleFields` will be `nil`, `delete` does a check and `ModuleFields` will have a `nil` assigment so everything keeps working. That's why I left it slim :smile:

Reply:
What happens if `event[mb.ModuleDataKey]` does not exist?

Reply:
That it will panic, of course, but `mb.ModuleDataKey` is hardcoded in the code that's why you don't need checking. I'll just add the checking, I don't think it's worth to talk too much about it but I have also received comments in other PR's about checking too much :smile:

Reply:
Check added